### PR TITLE
Update pluginwatcher doc

### PR DESCRIPTION
**Pull request by RenaudWasTaken**
**What this PR does / why we need it**: Adds more documentation about the pluginwatcher process and development

**Which issue(s) this PR fixes** See issue https://github.com/kubernetes/features/issues/595 and issue https://github.com/kubernetes/kubernetes/issues/65773

**Special notes for your reviewer**:
/sig node
/kind documentation
/cc @vikaschoudhary16 @figo @sbezverk @saad-ali @vladimirvivien @jiayingz 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/util/pluginwatcher/README [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+Once Kubelet determines that it is ready to use your plugin it will issue a
+Registration.NotifyRegistrationStatus gRPC call.
+
+
+## gRPC Service Overview
 
-- Implements the Registration service specified in
-  pkg/kubelet/apis/pluginregistration/v*/api.proto.
+Here are the general rules that Kubelet plugin developers should follow:
+- Run as 'root' user. Currently creating socket under PluginsDir, a root owned
```
Comment (vladimirvivien) [inline pkg/kubelet/util/pluginwatcher/README]: Is that `Run plugin as root` ?

**Review by vladimirvivien — COMMENTED** [review]
Added couple of suggestions.

File: pkg/kubelet/util/pluginwatcher/README (hunk 1 above)
Comment (RenaudWasTaken) [reply pkg/kubelet/util/pluginwatcher/README]: Well your plugin should have the correct permissions to create and read any files in the plugin directory.

File: pkg/kubelet/util/pluginwatcher/README [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
-Here are the general rules that Kubelet plugin developers should follow:
-- Run as 'root' user. Currently creating socket under PluginsSockDir, a root owned directory, requires
-  plugin process to be running as 'root'.
+Plugins are expected to implement the gRPC registration service specified in
+pkg/kubelet/apis/pluginregistration/v*/api.proto.
+
+
+## gRPC Service Lifecycle
+
+For any discovered plugin, kubelet will issue a Registration.GetInfo gRPC call
```
[the remark this answers is not in the document]
Comment (RenaudWasTaken) [reply pkg/kubelet/util/pluginwatcher/README]: Added

**Review by vladimirvivien — COMMENTED** [review]
Rename file README.md so mark down will render properly.

**Comment by RenaudWasTaken** [comment]
@vladimirvivien done!

File: pkg/kubelet/util/pluginwatcher/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+# Kubelet Interface
+
+For any kubelet components using the pluginwatcher module, you will need to
+implement the PluginHandler interface defined in the types.go file.
+
+The interface is documented and the implementations are registered with the
+pluginwatcher module in kubelet.go by calling AddHandler(pluginType, handler).
+
+
+The lifecycle follows a simple Validate -> Register -> DeRegister state machine.
```
Comment (figo) [inline pkg/kubelet/util/pluginwatcher/README.md]: the ascii graph is off when render the file, add a extra blank line will work (find out by rendering a readme file in my local IDE).

**Comment by RenaudWasTaken** [comment]
Fixed @figo

**Comment by nikhita** [comment]
/assign @yujuhong
for approval

**Comment by RenaudWasTaken** [comment]
ping @vishh for approval

**Comment by RenaudWasTaken** [comment]
Maybe @saad-ali can help with approval on this docs PR :) ?

**Comment by vishh** [comment]
Appreciate taking time to write this readme @RenaudWasTaken. Can you also add this text to the k8s device plugin docs page, or at-least link to this doc?

/lgtm
/approve

**Comment by RenaudWasTaken** [comment]
Thanks @vishh !

Yep, I'll take care of the PR tomorrow!

**Pull request merged**